corehq/apps/cloudcare/views.py

```diff
@@ -164,7 +164,7 @@ def set_cookie(response):  # overwrite the default noop set_cookie
             ).run()
             if login_as_users.total == 1:
                 def set_cookie(response):
-                    response.set_cookie(cookie_name, user.raw_username)
+                    response.set_cookie(cookie_name, urllib.parse.quote(user.raw_username))
```

Comment:
Does the bit reading from the cookie already reverse the URL serialization properly?  (Ie, it's going to see `+`, not `%2B` where the username gets looked up)

Reply:
Yes, it's read at [getRestoreAsUser](https://github.com/dimagi/commcare-hq/blob/bf313afaf838312250a087dffde3f706ca7cf75b/corehq/apps/cloudcare/static/cloudcare/js/formplayer/users/utils.js#L49-L51) via [jquery cookie library ](https://github.com/carhartl/jquery-cookie?tab=readme-ov-file#raw)which by default decodes on read.